### PR TITLE
Feature/fields group

### PR DESCRIPTION
**Pull request by DovydasNavickas**
https://github.com/SimplrJS/simplr-forms/issues/37

### CODE REVIEW

File: packages/simplr-forms-dom/src/components/fields-array.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { BaseFieldsArray } from "simplr-forms";
+
+export interface FieldsArrayProps extends CoreProps {
+}
+
+export interface FieldsArrayState extends CoreState {
+}
+
+interface Dictionary {
+    [key: string]: any;
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/fields-array.tsx]: Any?

File: packages/simplr-forms-dom/src/components/fields-array.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+export class FieldsArray extends BaseFieldsArray<FieldsArrayProps, FieldsArrayState> {
+    public Element: HTMLDivElement;
+
+    private setElementRef = (element: HTMLDivElement) => {
+        this.Element = element;
+    }
+
+    protected HTMLProps() {
+        let notHTMLProps = ["name", "index", "destroyOnUnmount"];
+        let newProps: { [id: string]: any } = {};
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/fields-array.tsx]: `Dictionary` type from L15 ?

File: packages/simplr-forms-dom/src/components/fields-group.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+export class FieldsGroup extends BaseFieldsGroup<FieldsGroupProps, FieldsGroupState> {
+    public Element: HTMLDivElement;
+
+    private setElementRef = (element: HTMLDivElement) => {
+        this.Element = element;
+    }
+
+    protected HTMLProps() {
+        let notHTMLProps = ["name", "destroyOnUnmount"];
+        let newProps: { [id: string]: any } = {};
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/fields-group.tsx]: `Dictionary` type from L14 ?

File: packages/simplr-forms/__tests__/abstractions/base-fields-group.test.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+describe("Fields Group Base", () => {
+    xit("throws when registering an already existing id", () => {
+
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms/__tests__/abstractions/base-fields-group.test.tsx]: Throw `not implemented`.

File: packages/simplr-forms-dom/src/components/fields-group.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export class FieldsGroup extends BaseFieldsGroup<FieldsGroupProps, FieldsGroupState> {
+    public Element: HTMLDivElement;
+
+    private setElementRef = (element: HTMLDivElement) => {
+        this.Element = element;
+    }
+
+    protected HTMLProps() {
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/fields-group.tsx]: Add missing return type (`Dictionary`).

File: packages/simplr-forms-dom/src/components/fields-group.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+interface Dictionary {
+    [key: string]: any;
+}
+
+export class FieldsGroup extends BaseFieldsGroup<FieldsGroupProps, FieldsGroupState> {
+    public Element: HTMLDivElement;
+
+    private setElementRef = (element: HTMLDivElement) => {
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms-dom/src/components/fields-group.tsx]: Missing return type.

File: packages/simplr-forms/src/actions/form-store.ts [hunk 7]
```diff
@@ -3,27 +3,35 @@ import { FieldValue } from "../contracts/field";
...
+        return this.fieldsGroupId;
+    }
+}
+
+export class FieldsArrayRegistered {
+    constructor(private fieldsGroupId: string) { }
 
-    public get InitialValue() {
-        return this.initialValue;
+    public get FieldsArrayId() {
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms/src/actions/form-store.ts]: Return types.

File: packages/simplr-forms/src/contracts/fields-array.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+export interface FieldsArrayProps {
+    name: string;
+    index: number;
+}
+
+export interface FieldsArrayState {
+}
+
+export interface FieldsArrayPropsObject { }
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms/src/contracts/fields-array.ts]: Add new line.

Comment (GiedriusGrabauskas) [inline packages/simplr-forms/src/contracts/fields-array.ts]: Or remove all new lines 😄

File: packages/simplr-forms/__tests__/abstractions/base-fields-group.test.tsx (hunk 4 above)
Comment (DovydasNavickas) [inline packages/simplr-forms/__tests__/abstractions/base-fields-group.test.tsx]: `xit` skips the test 👍

File: packages/simplr-forms-dom/src/components/fields-group.tsx (hunk 5 above)
Comment (DovydasNavickas) [inline packages/simplr-forms-dom/src/components/fields-group.tsx]: It's not dictionary anymore. Check the updated version.

File: packages/simplr-forms-dom/src/components/fields-array.tsx (hunk 1 above)
Comment (DovydasNavickas) [inline packages/simplr-forms-dom/src/components/fields-array.tsx]: What do you suggest when it's really anything there?

**Pull request merged**